.github/files/gh-autorelease/files/autorelease.sh

```diff
@@ -48,76 +58,80 @@ echo "::group::Creating $SLUG.zip"
 echo "::endgroup::"
 
-## Create the release note.
-# Extract the changelog section.
-echo "::group::Extracting release notes"
-if [[ ! -f CHANGELOG.md ]]; then
-	echo '::endgroup::'
-	echo '::error::No CHANGELOG.md for release notes.'
-	exit 1
-fi
-SCRIPT="
-	/^## \\[?$(sed 's/[.\[\]\\*^$\/()+?{}|]/\\&/g' <<<"${TAG#v}")\\]? - / {
-		bc
+if [[ -z "$ROLLING_MODE" ]]; then
+	## Create the release note.
+	# Extract the changelog section.
+	echo "::group::Extracting release notes"
+	if [[ ! -f CHANGELOG.md ]]; then
+		echo '::endgroup::'
+		echo '::error::No CHANGELOG.md for release notes.'
+		exit 1
+	fi
+	SCRIPT="
+		/^## \\[?$(sed 's/[.\[\]\\*^$\/()+?{}|]/\\&/g' <<<"${TAG#v}")\\]? - / {
+			bc
+			:a
+			n
+			/^## / {
+				q
+			}
+			:c
+			s/^## \[([^]]+)\]/## \1/
+			p
+			ba
+		}
+	"
+	ENTRY=$(sed -n -E -e "$SCRIPT" CHANGELOG.md)
+	if [[ -z "$ENTRY" ]]; then
+		echo '::endgroup::'
+		echo "::error::Failed to find section for ${TAG#v} in CHANGELOG.md"
+		exit 1
+	fi
+
+	# Strip unwanted sections.
+	SCRIPT="
 		:a
-		n
-		/^## / {
-			q
+		/^### .* This section will not be copied to readme\.txt/ {
+			:b
+			n
+			/^#/ ba
+			bb
 		}
-		:c
-		s/^## \[([^]]+)\]/## \1/
 		p
-		ba
-	}
-"
-ENTRY=$(sed -n -E -e "$SCRIPT" CHANGELOG.md)
-if [[ -z "$ENTRY" ]]; then
-	echo '::endgroup::'
-	echo "::error::Failed to find section for ${TAG#v} in CHANGELOG.md"
-	exit 1
+	"
+	ENTRY=$(sed -n -E -e "$SCRIPT" <<<"$ENTRY")
+
+	echo "Release notes:"
+	echo "-----"
+	echo "$ENTRY"
+	echo "-----"
+	echo "::endgroup::"
+else
+	## Using a brief explanation for the rolling release note.
+	ENTRY="### Rolling release based on the trunk branch."
+fi
+
+if [[ -n "$ROLLING_MODE" ]]; then
+	echo "::group::Deleting stale rolling release"
+
+	for R in $( gh release list --limit 100 --json tagName --jq '.[].tagName | select( contains( "rolling" ) )' ); do
+		echo "Found $R, deleting"
+		gh release delete "$R" --cleanup-tag --yes
+	done
+
+	echo "::endgroup::"
 fi
 
-# Strip unwanted sections.
-SCRIPT="
-	:a
-	/^### .* This section will not be copied to readme\.txt/ {
-		:b
-		n
-		/^#/ ba
-		bb
-	}
-	p
-"
-ENTRY=$(sed -n -E -e "$SCRIPT" <<<"$ENTRY")
-
-echo "Release notes:"
-echo "-----"
-echo "$ENTRY"
-echo "-----"
-echo "::endgroup::"
 
 echo "::group::Creating release"
-curl -v -L \
-	--write-out '%{response_code}' \
-	--output out.json \
-	--request POST \
-	--header "authorization: Bearer $API_TOKEN_GITHUB" \
-	--header 'content-type: application/json' \
-	--header 'accept: application/vnd.github.v3+json' \
-	--url "${GITHUB_API_URL}/repos/${GITHUB_REPOSITORY}/releases" \
-	--data "$(jq -n --arg tag "$TAG" --arg sha "$GITHUB_SHA" --arg title "$TITLE" --arg body "$ENTRY" '{ tag_name: $tag, target_commitish: $sha, name: $title, body: $body}')" \
-	2>&1 > code.txt
-cat out.json
-echo
-[[ "$(<code.txt)" =~ ^2[0-9][0-9]$ ]] || exit 1
+gh release create "$TAG" \
+	--notes "$ENTRY" \
+	--target "$GITHUB_SHA" \
+	--title "$TITLE"
```

Comment:
I happened to notice something interesting in the docs, `gh release create` can take the artifact directly instead of having to follow up with `gh release upload`.
```suggestion
gh release create "$TAG" "$SLUG.zip" \
	--notes "$ENTRY" \
	--target "$GITHUB_SHA" \
	--title "$TITLE"
```
(then delete the group calling `gh release upload` below)